src/poseidon/mod.rs

```diff
@@ -313,11 +313,29 @@ impl<F: PrimeField> CryptographicSponge for PoseidonSponge<F> {
         bits
     }
 
+    fn squeeze_field_elements_with_sizes<F2: PrimeField>(
+        &mut self,
+        sizes: &[FieldElementSize],
+    ) -> Vec<F2> {
+        if F::characteristic() == F2::characteristic() {
+            // native case
+            let mut buf = Vec::with_capacity(sizes.len());
+            batch_field_cast(
+                &self.squeeze_native_field_elements_with_sizes(sizes),
+                &mut buf,
+            )
+            .unwrap();
+            buf
+        } else {
+            squeeze_field_elements_with_sizes_default_impl(self, sizes)
+        }
+    }
+
     fn squeeze_field_elements<F2: PrimeField>(&mut self, num_elements: usize) -> Vec<F2> {
         if TypeId::of::<F>() == TypeId::of::<F2>() {
             let result = self.squeeze_native_field_elements(num_elements);
             let mut casted = Vec::with_capacity(result.len());
-            batch_field_cast(&result, &mut casted);
+            batch_field_cast(&result, &mut casted).unwrap();
```

Comment:
Small nit, the past tense of "cast" is "cast" =)

Reply:
Ah thanks for the catch! (Oh no I've been using the word `casted` for a long time without realizing it)